src/main/java/io/vertx/ext/mongo/BulkOperation.java

```diff
@@ -252,23 +256,24 @@ public BulkOperation setMulti(boolean multi) {
   public boolean equals(Object o) {
     if (this == o) return true;
     if (o == null || getClass() != o.getClass()) return false;
-
-    BulkOperation other = (BulkOperation) o;
-
-    if (type != other.type) return false;
-    if (filter != null ? !filter.equals(other.filter) : other.filter != null) return false;
-    if (document != null ? !document.equals(other.document) : other.document != null) return false;
-
-    return upsert == other.upsert && multi == other.multi;
+    BulkOperation operation = (BulkOperation) o;
+    return isUpsert() == operation.isUpsert() && isMulti() == operation.isMulti() && getType() == operation.getType() && Objects.equals(getFilter(), operation.getFilter()) && Objects.equals(getDocument(), operation.getDocument()) && Objects.equals(getCollation(), operation.getCollation());
   }
 
   @Override
   public int hashCode() {
-    int result = type.hashCode();
-    result = 31 * result + filter.hashCode();
-    result = 31 * result + document.hashCode();
-    result = 31 * result + (upsert ? 1 : 0);
-    result = 31 * result + (multi ? 1 : 0);
-    return result;
+    return Objects.hash(getType(), getFilter(), getDocument(), isUpsert(), isMulti(), getCollation());
```

Comment:
Same here: why getters?

Reply:
This is auto generated code (I will change it :disappointed: )